client/my-sites/marketing/tools/google-adwords.tsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import page from 'page';
+import PropTypes from 'prop-types';
+import React, { Fragment, FunctionComponent } from 'react';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { addQueryArgs } from 'lib/url';
+import Button from 'components/button';
+import { getSelectedSite } from 'state/ui/selectors';
+import GoogleVoucherDetails from 'my-sites/checkout/checkout-thank-you/google-voucher';
+import { isPremium, isBusiness, isEcommerce, isEnterprise } from 'lib/products-values';
+import isSiteAtomic from 'state/selectors/is-site-automated-transfer';
+import MarketingToolsFeature from './feature';
+import { PLAN_PREMIUM } from 'lib/plans/constants';
+import QuerySiteVouchers from 'components/data/query-site-vouchers';
+import { recordTracksEvent as recordTracksEventAction } from 'state/analytics/actions';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface ConnectedProps {
+	isAtomic: boolean;
+	isJetpack: boolean;
+	isPremiumOrHigher: boolean;
+	recordTracksEvent: () => void;
+	siteId: number | null;
+	siteSlug: string | null;
+}
+
+export const MarketingToolsGoogleAdwordsFeature: FunctionComponent< ConnectedProps > = ( {
+	isAtomic,
+	isJetpack,
+	isPremiumOrHigher,
+	recordTracksEvent,
+	siteId,
+	siteSlug,
+} ) => {
+	const translate = useTranslate();
+	if ( ( isJetpack && ! isAtomic ) || ! siteId || ! siteSlug ) {
+		return null;
+	}
+
+	const handleUpgradeClick = () => {
+		recordTracksEvent( 'calypso_marketing_tools_adwords_plan_upgrade_button_click' );
+		page( addQueryArgs( { plan: PLAN_PREMIUM }, `/plans/${ siteSlug }` ) );
+	};
+
+	const renderButton = () => {
+		if ( isPremiumOrHigher ) {
+			return <GoogleVoucherDetails />;
+		}
+		return <Button onClick={ handleUpgradeClick }>{ translate( 'Upgrade To Premium' ) }</Button>;
+	};
+
+	return (
+		<Fragment>
+			<QuerySiteVouchers siteId={ siteId } />
+			<MarketingToolsFeature
+				title={ translate( 'Advertise with your $100 Google Adwords credit' ) }
+				description={ translate(
+					"Advertise your site where most people are searching: Google. You've got a $100 credit with Google Adwords to drive traffic to your most important pages."
+				) }
+				imagePath="/calypso/images/illustrations/marketing.svg"
+			>
+				{ renderButton() }
```

Comment:
is there a strong reason not to inline this? our preference is inlining React components except in the cases where the display logic is most complicated, and even then we've tended to see refactors in order to bring together the JSX

just hard to bounce around in the file with `renderThing()`

is this pattern something you feel is common?

Reply:
I usually do this when the complexity starts creeping in. Sorta like breaking out functions. In this case I do not see a reason to break it out.